torch/csrc/Exceptions.h

```diff
@@ -138,4 +157,27 @@ struct ValueError : public PyTorchError {
   }
 };
 
+// ATen warning handler for Python
+struct PyWarningHandler{
+public:
+  using warning_buffer_t =
+    std::queue<std::pair<c10::SourceLocation, std::string>>;
+
+  static void py_warning_handler(
+    const c10::SourceLocation& source_location,
+    const std::string& msg);
+
+  static warning_buffer_t warning_buffer;
+};
+
+struct EnforceWarningBuffer {
+public:
+/// See NOTE [ Conversion Cpp Python Warning ] for noexcept justification
+  EnforceWarningBuffer() noexcept(true);
+  ~EnforceWarningBuffer() noexcept(false);
```

Comment:
Wow, TIL this is allowed hahaha

Reply:
This is my attempt to make sure that we don't get killed by throwing two exceptions.